lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/StakePools.hs

```diff
@@ -126,39 +133,51 @@ spec = do
             unsafeRequest @[ApiStakePool] ctx listStakePoolsEp Empty
         w <- emptyWallet ctx
         r <- joinStakePool ctx p (w, "Secure Passphrase")
-        expectResponseCode HTTP.status501 r
+        expectResponseCode HTTP.status403 r
 
     it "STAKE_POOLS_JOIN_01 - I cannot join another until I quit\
         \ or maybe I will just re-join another?" $ \ctx -> do
         (_, p1:p2:_) <- eventually $
             unsafeRequest @[ApiStakePool] ctx listStakePoolsEp Empty
         w <- emptyWallet ctx
         r1 <- joinStakePool ctx p1 (w, "Secure Passprase")
-        expectResponseCode HTTP.status501 r1
+        expectResponseCode HTTP.status403 r1
 
         r2 <- joinStakePool ctx p2 (w, "Secure Passprase")
-        expectResponseCode HTTP.status501 r2
+        expectResponseCode HTTP.status403 r2
 
     it "STAKE_POOLS_JOIN_01 - \
         \I definitely can quit and join another" $ \ctx -> do
         (_, p1:p2:_) <- eventually $
             unsafeRequest @[ApiStakePool] ctx listStakePoolsEp Empty
         w <- emptyWallet ctx
         r1 <- joinStakePool ctx p1 (w, "Secure Passprase")
-        expectResponseCode HTTP.status501 r1
+        expectResponseCode HTTP.status403 r1
 
         r1q <- quitStakePool ctx p1 (w, "Secure Passprase")
         expectResponseCode HTTP.status501 r1q
 
         r2 <- joinStakePool ctx p2 (w, "Secure Passprase")
-        expectResponseCode HTTP.status501 r2
+        expectResponseCode HTTP.status403 r2
+
+    it "STAKE_POOLS_JOIN_01 - Cannot join non-existant stakepool" $ \ctx -> do
+        let pId =
+                "0000000000000000000000000000000000000000000000000000000000000000"
+        let (Right addr) = fromHex pId
+        let poolIdAbsent = ApiT $ PoolId addr
+        let poolMetrics = Types.ApiStakePoolMetrics (Quantity 0) (Quantity 0)
+        let pool = Types.ApiStakePool poolIdAbsent poolMetrics 0
+        w <- emptyWallet ctx
+        r <- joinStakePool ctx pool (w, "Secure Passphrase")
```

Comment:
**[change required]**
`joinStakePool` requires a full `ApiStakePool` as argument which forces to create dummy values. I'd suggest to either:

1. drop `joinStakePool` and simply use the `request` combinator from the integration framework. 
2. Review `joinStakePool` and `joinStakePoolEp` to only require a stake pool id instead of a whole body

**Why**
The goal of the integration DSL is to favor readability and strive to have readable integration scenarios. By adding extra non-orthogonal (a.k.a overlapping) combinators, we do actually create noise and reduce readability and maintenance in the long run. Here, we are forced to create a fake pool metrics and a fake `ApiStakePool` object, whereas, we really just need to write:

```hs
    let poolId = T.replicate 64 "0"
    w <- emptyWallet ctx
    let payload = Json [aesonQQ| { "passphrase": "Secure Passphase" } |]
    request @(ApiTransaction 'Testnet) ctx (joinStakePoolEp poolId w) Default payload
```

Reply:
I have spotted this, and planned to review it in forthcoming PR (here just used the same way as in other examples in the test suite). Added a point in development plan : 
```
[ ] review joinStakePoolEp in DSL
```
to tackle this in #896